nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+          Further nginx configuration can be done by adapting <literal>services.nginx.virtualHosts.&lt;name&gt;</literal>.
+        '';
+      };
+
+      https = mkOption {
+        type = bool;
+        default = true;
+        description = ''
+          Whether to use HTTPS. When nginx integration is enabled, this option forces SSL and enables ACME.
+          Please note that Sympa web interface always uses https links even when this option is disabled.
+        '';
+      };
+
+      fcgiProcs = mkOption {
+        type = ints.positive;
+        default = 2;
+        description = "Number of FastCGI processes to fork.";
+      };
+    };
+
+    mta = {
+      type = mkOption {
+        type = enum [ "postfix" "none" ];
+        default = "postfix";
+        description = ''
+          Mail transfer agent (MTA) integration. Use `none` if you want to configure it yourself.
+
+          The `postfix` integration sets up local Postfix instance that will pass incoming
+          messages from configured domains to Sympa. You still need to configure at least
+          outgoing message handling using e.g. <option>services.postfix.relayHost</option>.
+        '';
+      };
+    };
+
+    settings = mkOption {
+      type = attrsOf (either (either str int) bool);
+      default = {};
+      example = {
+        default_home = "lists";
+        viewlogs_page_size = 50;
+      };
+      description = ''
+        The sympa.conf configuration file as key value set.
+      '';
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    services.sympa.settings = (mapAttrs (_: v: mkDefault v) {
+      domain     = if cfg.mainDomain != null then cfg.mainDomain else (head fqdns);
+      listmaster = concatStringsSep "," cfg.listMasters;
+      lang       = cfg.lang;
+
+      home       = "${dataDir}/list_data";
+      arc_path   = "${dataDir}/arc";
+      bounce_path = "${dataDir}/bounce";
+
+      sendmail = "/run/wrappers/bin/sendmail";
+
+      db_type = cfg.database.type;
+      db_name = cfg.database.name;
+      db_passwd = "#dbpass#";
+    }
+    // (optionalAttrs (cfg.database.host != null) {
+      db_host = cfg.database.host;
+    })
+    // (optionalAttrs (cfg.database.port != null) {
+      db_port = cfg.database.port;
+    })
+    // (optionalAttrs (cfg.database.user != null) {
+      db_user = cfg.database.user;
+    })
+    // (optionalAttrs (cfg.mta.type == "postfix") {
+      sendmail_aliases = "${dataDir}/sympa_transport";
+      aliases_program  = "${pkgs.postfix}/bin/postmap";
+      aliases_db_type  = "hash";
+    })
+    // (optionalAttrs cfg.web.enable {
+      static_content_path = "${dataDir}/static_content";
+      css_path            = "${dataDir}/static_content/css";
+      pictures_path       = "${dataDir}/static_content/pictures";
+      mhonarc             = "${pkgs.perlPackages.MHonArc}/bin/mhonarc";
+    }));
+
+    environment = {
+      systemPackages = [ pkg ];
```

Comment:
Does it make sense for this need to be installed in the users environment? If not this can be removed

Reply:
There's `sympa.pl` which can be used for list administration. Also manual pages.